### DISCUSSION TITLE
Is it possible to obtain ensemble of protein structures through CryoDRGN?

### DISCUSSION THREAD

**Discussion by Immortals-33** [zhonge/cryodrgn#398] [Q&A] [answered]
Dear authors:

Thanks you for this great work and making it open-source!

I have a simple question, which might be confusing from your point of view: **Is there a way to obtain multiple conformations of macromolecular structures using CryoDRGN?**

From my little knowledge on this field, CryoDRGN utilizes 2D images obtained from Cryo-EM and construct the heterogenity of those images, and then output *k* density maps. By then we can build the protein model manually or use some automated modeling pipeline like [*ModelAngelo*](https://www.nature.com/articles/s41586-024-07215-4) to get reasonable PDB structures with spatially well-assigned main chains and side chains conformations.

It seems CryoDRGN is mainly utilized to improve the resolution of density maps for a purpose of local refinement, yet can I treat these multiple density maps as **different conformations** and use them to construct conformational ensemble of a single protein? Or can I reach the same place by conformational information from landscape analysis by CryoDRGN? If this is possible in practice, do you consider these conformation reasonable or explainable?

This might be a fundamental question as I'm a newcomer and not very familiar with these pipelines. I would be much appreciated if you provide some insights upon this topic.

**Comment by michal-g** [comment, marked as answer]
Hi, I moved this thread here as it would be better suited as a Discussion than an Issue — but in short, yes, one of the key properties of cryoDRGN is that it can recover different conformations of the same molecule appearing in a dataset, as well as different molecular structures appearing in the same dataset. We refer to these two types of heterogeneity as conformational and compositional respectively, and indeed encourage you to use tools such as `cryodrgn train_vae` and `cryodrgn abinit_het` that model the heterogeneity of the dataset using a latent embedding space (compared to `train_nn` and `abinit_homo` which train a homogeneous model). All three of `cryodrgn analyze`, `cryodrgn analyze_landscape`, and `cryodrgn analyze_landscape_full` can be applied to a trained `train_vae` or `abinit_het` model to analyze the heterogeneity it has uncovered at successively deeper levels of complexity.

Whether or not these conformations end up being reasonable or explainable is of course context-dependent, but you can find some good examples and discussion of where they end up being both in the [original cryoDRGN paper](https://www.nature.com/articles/s41592-020-01049-4), in the sections starting with _CryoDRGN models both discrete and continuous structural heterogeneity_. Note that "discrete/continuous heterogeneity" can be used interchangeably with "compositional/conformational heterogeneity".

Hope this helps!

**Reply by Immortals-33** [reply]
Hi Michal,

Thanks for your reply! It provides me with a better understanding on how it works.

I'll try out modeling heterogenity ensembles with CryoDRGN and dive deeper into it. Hopefully I'll share come back with some interesting results another day.
